src/main/java/commands/Command.java

```diff
@@ -0,0 +1,141 @@
+            String[] inputs = message.split(" ");
+            int index = Integer.parseInt(inputs[1]);
+            tasks.markTaskAsDone(index);
+            storage.saveTasks(tasks);
+        }
+    },
+    /**
+     * Command to unmark a specified task as done.
+     */
+    UNMARK {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            // Extract task index from argument and mark the task as not done
+            String[] inputs = message.split(" ");
+            int index = Integer.parseInt(inputs[1]);
+            tasks.unmarkTaskAsDone(index);
+            storage.saveTasks(tasks);
+        }
+    },
+    /**
+     * Command to add a todo task to tasklist.
+     */
+    ADD_TODO {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            // Create a new tasks.ToDo task and add it to the task list
+            Task task = tasks.initTask(message, "todo");
+            tasks.addTasktoTaskList(task);
+            ui.triggerAddMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+    /**
+     * Command to add a deadline task to tasklist.
+     */
+    ADD_DEADLINE {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            // Create a new tasks.Deadline task and add it to the task list
+            Task task = tasks.initTask(message, "deadline");
+            tasks.addTasktoTaskList(task);
+            ui.triggerAddMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+    /**
+     * Command to add a event task to tasklist.
+     */
+    ADD_EVENT {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            // Create a new tasks.Event task and add it to the task list
+            Task task = tasks.initTask(message, "event");
+            tasks.addTasktoTaskList(task);
+            ui.triggerAddMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+
+    /**
+     * Command to delete a task from tasklist by index.
+     */
+    DELETE {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            String[] inputs = message.split(" ");
+            int index = Integer.parseInt(inputs[1]);
+            Task task = tasks.removeTaskfromTaskList(index);
+            ui.triggerDeleteMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+    FIND {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            String[] inputs = message.split(" ");
+            String queryString = inputs[1];
+            ArrayList<Task> tempList = new ArrayList<>();
+            TaskList tempTaskList = tasks.filter(queryString);
+            tempTaskList.yapTasks();
+        }
+    }
```

Comment:
It is a very interesting way to handle different Command Type. It is good that all the constant names are uppercase which is very clear to other developer/reviewer.